src/components/ExpensiTextInput/index.ios.js

```diff
@@ -0,0 +1,126 @@
+            labelTranslateX: new Animated.Value(hasValue
+                ? ACTIVE_LABEL_TRANSLATE_X(props.translateX) : INACTIVE_LABEL_TRANSLATE_X),
+            labelScale: new Animated.Value(hasValue ? ACTIVE_LABEL_SCALE : INACTIVE_LABEL_SCALE),
+        };
+
+        this.input = null;
+    }
+
+    animateLabel = (translateY, translateX, scale) => {
+        Animated.parallel([
+            Animated.spring(this.state.labelTranslateY, {
+                toValue: translateY,
+                duration: 80,
+                useNativeDriver: true,
+            }),
+            Animated.spring(this.state.labelTranslateX, {
+                toValue: translateX,
+                duration: 80,
+                useNativeDriver: true,
+            }),
+            Animated.spring(this.state.labelScale, {
+                toValue: scale,
+                duration: 80,
+                useNativeDriver: true,
+            }),
+        ]).start();
+    }
+
+    onFocus = () => {
+        if (this.props.onFocusExtra) { this.props.onFocusExtra(); }
+        this.setState({isFocused: true});
+        if (this.props.value.length === 0) {
+            this.animateLabel(
+                ACTIVE_LABEL_TRANSLATE_Y,
+                ACTIVE_LABEL_TRANSLATE_X(this.props.translateX),
+                ACTIVE_LABEL_SCALE,
+            );
+        }
+    }
+
+    onBlur = () => {
+        if (this.props.onBlurExtra) { this.props.onBlurExtra(); }
+        this.setState({isFocused: false});
+        if (this.props.value.length === 0) {
+            this.animateLabel(INACTIVE_LABEL_TRANSLATE_Y, INACTIVE_LABEL_TRANSLATE_X, INACTIVE_LABEL_SCALE);
+        }
+    }
+
+    focus = () => this.input.focus()
+
+    render() {
+        const {
+            label, value, placeholder, error, containerStyles, androidStyles, ...inputProps
+        } = this.props;
+        const {
+            isFocused, labelTranslateY, labelTranslateX, labelScale,
+        } = this.state;
+
+        const hasLabel = !!label.length;
+        return (
+            <ExpensiTextInputWrapper
+                containerStyles={containerStyles}
+                isFocused={isFocused}
+                error={error}
+                onPress={this.focus}
+                hasLabel={hasLabel}
+            >
+                <>
+                    {hasLabel ? (
+                        <ExpensiTextInputLabel
+                            label={label}
+                            labelTranslateX={labelTranslateX}
+                            labelTranslateY={labelTranslateY}
+                            labelScale={labelScale}
+                        />
+                    ) : null}
+                    <TextInput
+                        ref={ref => this.input = ref}
+                        value={value}
+                        onFocus={this.onFocus}
+                        onBlur={this.onBlur}
+                        placeholder={isFocused || !label ? placeholder : null}
```

Comment:
This is confusing ... can you include parens to make it a bit clearer?

Reply:
Done